tests/test_configuration.py

```diff
@@ -0,0 +1,108 @@
+import json
+from os import environ
+from unittest.mock import MagicMock, patch
+
+import pytest
+
+from lbz.aws_ssm import SSM
+from lbz.configuration import ConfigValue, EnvValue, SSMValue
+from lbz.exceptions import ConfigurationMissingError, ConfigurationParsingError
+
+
+class MyConfigValue(ConfigValue):
+    def getter(self) -> str:
+        return "test-value"
+
+
+class MyNoneConfigValue(ConfigValue):
+    def getter(self) -> None:
+        return None
+
+
+class TestBaseConfig:
+    def test_if_getter_was_used(
+        self,
+    ) -> None:
+        cfg = MyConfigValue("key")
+
+        assert cfg.value == "test-value"
+
+    def test_returns_default_when_no_value_set(self) -> None:
+        cfg = MyNoneConfigValue("test", default=42, parser=int)
+
+        assert cfg.value == 42
+
+    def test_raises_missing_congiration_when_value_is_none(self) -> None:
+        cfg = MyNoneConfigValue("test")
+
+        with pytest.raises(ConfigurationMissingError, match="'test' was not defined."):
+            cfg.value  # pylint: disable=pointless-statement
+
+    def test_if_parser_was_used(self) -> None:
+        class MyIntConfigValue(ConfigValue):
+            def getter(self) -> int:
+                return 1
+
+        cfg = MyIntConfigValue("key")
```

Comment:
I would rather use a custom parser here in this test to show more directly what you expect.

Now, you rely on a default behavior that is tested multiple times in other places.